PerformanceTransactionTest.testConsumeTxnMessage

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Example failure

https://github.com/codelipenghui/incubator-pulsar/actions/runs/3368394689/jobs/5587018417

### Exception stacktrace

```
  Error:  Tests run: 10, Failures: 1, Errors: 0, Skipped: 9, Time elapsed: 102 s <<< FAILURE! - in org.apache.pulsar.testclient.PerformanceTransactionTest
8752
  Error:  testConsumeTxnMessage(org.apache.pulsar.testclient.PerformanceTransactionTest)  Time elapsed: 6.199 s  <<< FAILURE!
8753
  java.lang.AssertionError: expected [null] but found [org.apache.pulsar.client.impl.MessageImpl@17fba246]
8754
  	at org.testng.Assert.fail(Assert.java:99)
8755
  	at org.testng.Assert.failNotSame(Assert.java:1033)
8756
  	at org.testng.Assert.assertNull(Assert.java:965)
8757
  	at org.testng.Assert.assertNull(Assert.java:953)
8758
  	at org.apache.pulsar.testclient.PerformanceTransactionTest.testConsumeTxnMessage(PerformanceTransactionTest.java:246)
8759
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
8760
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
8761
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
8762
  	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
8763
  	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
8764
  	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
8765
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
8766
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
8767
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
8768
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
8769
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
8770
  	at java.base/java.lang.Thread.run(Thread.java:833)
```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/pulsar-testclient/src/test/java/org/apache/pulsar/testclient/PerformanceTransactionTest.java
+++ b/pulsar-testclient/src/test/java/org/apache/pulsar/testclient/PerformanceTransactionTest.java
@@ -198,7 +198,7 @@ public class PerformanceTransactionTest extends MockedPulsarServiceBaseTest {
 
     @Test
     public void testConsumeTxnMessage() throws Exception {
-        String argString = "%s -r 10 -u %s -txn -ss %s -st %s -sp %s -ntxn %d -tto 5";
+        String argString = "%s -u %s -txn -ss %s -st %s -sp %s -ntxn %d -tto 5";
         String subName = "sub";
         String topic = testTopic + UUID.randomUUID();
         String args = String.format(argString, topic, pulsar.getBrokerServiceUrl(), subName,

--- a/pulsar-testclient/src/test/java/org/apache/pulsar/testclient/PerformanceTransactionTest.java
+++ b/pulsar-testclient/src/test/java/org/apache/pulsar/testclient/PerformanceTransactionTest.java
@@ -163,7 +163,7 @@ public class PerformanceTransactionTest extends MockedPulsarServiceBaseTest {
 
     @Test
     public void testProduceTxnMessage() throws InterruptedException, PulsarClientException {
-        String argString = "%s -r 10 -u %s -m %d -txn";
+        String argString = "%s -r 50 -u %s -m %d -txn";
         String topic = testTopic + UUID.randomUUID();
         int totalMessage = 100;
         String args = String.format(argString, topic, pulsar.getBrokerServiceUrl(), totalMessage);
@@ -174,7 +174,6 @@ public class PerformanceTransactionTest extends MockedPulsarServiceBaseTest {
                 .subscribe();
         Thread thread = new Thread(() -> {
             try {
-                log.info("");
                 PerformanceProducer.main(args.split(" "));
             } catch (Exception e) {
                 e.printStackTrace();
@@ -182,6 +181,13 @@ public class PerformanceTransactionTest extends MockedPulsarServiceBaseTest {
         });
         thread.start();
         thread.join();
+
+        Awaitility.await().untilAsserted(() -> {
+            admin.transactions().getCoordinatorStats().forEach((integer, transactionCoordinatorStats) -> {
+                Assert.assertEquals(transactionCoordinatorStats.ongoingTxnSize, 0);
+            });
+        });
+
         Consumer<byte[]> consumer = pulsarClient.newConsumer().subscriptionName("subName").topic(topic)
                 .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest)
                 .subscriptionType(SubscriptionType.Exclusive)
@@ -198,7 +204,7 @@ public class PerformanceTransactionTest extends MockedPulsarServiceBaseTest {
 
     @Test
     public void testConsumeTxnMessage() throws Exception {
-        String argString = "%s -u %s -txn -ss %s -st %s -sp %s -ntxn %d -tto 5";
+        String argString = "%s -r 50 -u %s -txn -ss %s -st %s -sp %s -ntxn %d -tto 5";
         String subName = "sub";
         String topic = testTopic + UUID.randomUUID();
         String args = String.format(argString, topic, pulsar.getBrokerServiceUrl(), subName,

